backend/kernelCI_app/views/hardwareView.py

```diff
@@ -0,0 +1,70 @@
+import datetime
+from django.http import JsonResponse
+from django.views import View
+from django.utils import timezone
+from collections import defaultdict
+from kernelCI_app.models import Tests
+
+
+class HardwareView(View):
+    def __getSlowResult(self, start_date):
+        tests = Tests.objects.filter(
+            start_time__gte=start_date, environment_compatible__isnull=False
+        ).values("environment_compatible", "status", "start_time", "build__valid")
+
+        hardware = {}
+        for test in tests:
+            for compatible in test["environment_compatible"]:
+                if hardware.get(compatible) is None:
+                    hardware[compatible] = {
+                        "hardwareName": compatible,
+                        "statusCount": defaultdict(int),
+                        "buildCount": {"valid": 0, "invalid": 0, "null": 0},
+                    }
+                if test["status"] is None:
+                    hardware[compatible]["statusCount"]["NULL"] += 1
+                else:
+                    hardware[compatible]["statusCount"][test["status"]] += 1
+
+                if test["build__valid"] is None:
+                    hardware[compatible]["buildCount"]["null"] += 1
+                elif test["build__valid"] is True:
+                    hardware[compatible]["buildCount"]["valid"] += 1
+                else:
+                    hardware[compatible]["buildCount"]["invalid"] += 1
+
+        result = {"hardware": list(hardware.values())}
+
+        return result
+
+    def __getFastResult(self, start_date):
+        hardwares = set()
+        tests = Tests.objects.filter(
+            start_time__gte=start_date, environment_compatible__isnull=False
+        ).values("environment_compatible", "start_time")
+
+        for test in tests:
+            for compatible in test["environment_compatible"]:
+                hardwares.add(compatible)
+
+        result = {"hardware": list(hardwares)}
+        return result
+
+    def get(self, request):
+        mode = request.GET.get("mode", "fast")
+        try:
+            daysInterval = int(request.GET.get("daysInterval", 3))
```

Comment:
I would either move this logic to a separate function as @lfjnascimento said or use the same variable name as it is in treeView and treeViewFast (interval_days for the variable and intervalInDays for the url parameter)

Reply:
Also, use a const for the default daysInterval instead of a magic number